crates/router/src/analytics_validator.rs

```diff
@@ -0,0 +1,25 @@
+use analytics::errors::AnalyticsError;
+use api_models::analytics::AnalyticsRequest;
+use common_utils::errors::CustomResult;
+use currency_conversion::types::ExchangeRates;
+use router_env::logger;
+
+use crate::core::currency::get_forex_exchange_rates;
+
+pub async fn request_validator(
+    req_type: AnalyticsRequest,
+    state: &crate::routes::SessionState,
+) -> CustomResult<Option<ExchangeRates>, AnalyticsError> {
+    let forex_enabled = state.conf.analytics.get_inner().get_forex_enabled();
+    let require_forex_functionality = req_type.requires_forex_functionality();
+    // other validation logic based on `req_type` goes here
```

Comment:
Do we want to keep this comment or remove it?

Reply:
We can remove it.